### ISSUE TITLE
The new rustup and cargo plugin are incredibly slow

### ISSUE THREAD

**Issue by QuarticCat** [ohmyzsh/ohmyzsh#10390]
### Describe the bug

After I update my plugins, I found the start time of my ZSH becomes extremely slow. I located the problem to these two plugins.

Here is my `.zshrc` file https://github.com/QuarticCat/dotfiles/blob/a9662d9114747f1080a2fca115a077273febfe24/dot_zshrc

If uncomment the `OMZP::cargo` line and the `OMZP::rustup` line, the start time will be 3 times slower.

### Steps to reproduce

Copy the content of these two plugins into files and test their execution time.

```console
$ hyperfine './test.sh'
```

### Expected behavior

Shouldn't cost too much time.

### Screenshots and recordings

The result of rustup plugin:

```test
Benchmark 1: ./test.sh
  Time (mean ± σ):      67.8 ms ±   6.4 ms    [User: 57.0 ms, System: 10.9 ms]
  Range (min … max):    62.3 ms …  84.6 ms    44 runs
```

The result of cargo plugin:

```
Benchmark 1: ./test.sh
  Time (mean ± σ):      66.7 ms ±   7.0 ms    [User: 56.6 ms, System: 10.0 ms]
  Range (min … max):    57.1 ms …  77.3 ms    49 runs
```

### OS / Distro

Arch Linux

### Zsh version

5.8

### Terminal emulator

yakuake

**Comment by QuarticCat** [comment]
It turns out to be the problem of `rustup --version`

```
Benchmark 1: rustup --version
  Time (mean ± σ):      67.4 ms ±  21.3 ms    [User: 54.2 ms, System: 8.7 ms]
  Range (min … max):    57.0 ms … 149.6 ms    19 runs
```

So I will close this issue.

**Comment by mcornella** [comment]
Fixed in #10392.
